packages/controller-utils/src/create-service-policy.ts

```diff
@@ -203,27 +207,21 @@ export function createServicePolicy({
   });
   const onBreak = circuitBreakerPolicy.onBreak.bind(circuitBreakerPolicy);
 
-  const onDegradedListeners: (() => void)[] = [];
+  const onDegradedEventEmitter = new CockatielEventEmitter<void>();
```

Comment:
Here we are reusing the Cockatiel built-in EventEmitter class instead of implementing the EventEmitter pattern ourselves. We do this because we want the signature of the `onDegraded` method to match the signatures for `onRetry` and `onBreak` (`onDegraded` is a custom hook we add, whereas `onRetry` and `onBreak` essentially come directly from Cockatiel).